format/index.md

```diff
@@ -113,9 +113,14 @@ of resources, or resource relationship primarily targeted by a request.
 
 A document **MUST** contain either primary data or an array of [error objects](#errors).
 
-Primary data **MUST** appear under a top-level key named `"data"`. Primary
-data **MUST** be either a single resource object, an array of resource
-objects, or a value representing a resource relationship.
+Primary data **MUST** appear under a top-level key named `"data"`.
+
+Primary data **MUST** be either:
+
+* a single resource object or `null`, for requests that target single resources
+* an array of resource objects or an empty array ([]), for requests that target
+resource collections
+* resource linkage, for requests that target a resource's relationship
```

Comment:
@bintoro what this change is meant to address is the fact that the current definition of primary data, as containing "a single resource object, an array of resource objects, or a value representing a resource relationship" isn't exhaustive--it excludes the null case, when null is being returned instead of a single resource (i.e. not as linkage). It also makes more explicit the possibility that the array can be empty.

It does mean that the sentences you quoted are somewhat less necessary, but I'd left them for now for clarity